sdk/python/tests/compiler/compiler_tests.py

```diff
@@ -697,13 +697,61 @@ def some_pipeline():
       task2 = some_op()
       task3 = some_op()
 
-      dsl.get_pipeline_conf().set_image_pull_policy(policy="Allways")
-      workflow_dict = compiler.Compiler()._compile(some_pipeline)
-      for template in workflow_dict['spec']['templates']:
-        container = template.get('container', None)
-        if container:
-          self.assertEqual(template['container']['imagePullPolicy'], "Allways")
+      dsl.get_pipeline_conf().set_image_pull_policy(policy="Always")
+    workflow_dict = compiler.Compiler()._compile(some_pipeline)
+    for template in workflow_dict['spec']['templates']:
+      container = template.get('container', None)
+      if container:
+        self.assertEqual(template['container']['imagePullPolicy'], "Always")
+
+  
+  def test_image_pull_policy_step_spec(self):
+    def some_op():
+      return dsl.ContainerOp(
+          name='sleep',
+          image='busybox',
+          command=['sleep 1'],
+      )
+
+    def some_other_op():
```

Comment:
I'm not sure you need another or.
You can just test on the second task created from `some_op`.
`set_image_pull_policy` is per-task, not per-componet.

Reply:
I added a second op in order to set the name so that I could filter them on template.name
```python 
if template['name' ] == "other":
```. I wanted to have a test that checked that it worked with both setting on pipeline level and on op without interfering with each other. As I understand that would not be possible if i dont create a new op?

Reply:
JFYI, there is `task.set_display_name(name)` and also `add_annotation`.

Reply:
Cool, thanks! I will leave it as it is for know.